### DISCUSSION TITLE
[REQUEST] Need a new row option called "prefix"

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#1628] [General]
### Before opening a feature request

- [X] I checked the *next* branch to see if the feature has already been implemented
- [X] I searched existing reports to see if it is already requested.

### What is the user problem or growth opportunity you want to see solved?

I'm a newcomer for rofi, and I was using ulauncher before.

Ulauncher has a feature called "SHORTCUTS", and it can quickly open a url like "https://www.google.com/search?q=xxxxxx"

I'm going to write a rofi script for the feature, but I'm confused about how to generate the url with rofi

### How do you know that this problem exists today? Why is this important?

I can easilly get the prefix of url (https://www.google.com/search?q=) with rofi feature "ROFI_INFO".

But there is a problem about how to deliver the prefix after I type the query string.

My solution is writing the prefix to a temp file, and read the prefix in the next execution of script.

In the branch "next", "ROFI_DATA" can help it.

And I have a new idea. If there is a row option called "prefix", the value of prefix should auomatically appear before cursor when select the row entry, and I could just type the query string to generate the url.

### Who will benefit from it?

it can benefit to resolve complex data when write a rofi script

### Rofi version (rofi -v)

1.7.3

### Configuration

all is default

**Comment by DaveDavenport** [comment]
This is more a how do I use, then an issue. If something needs to be added, Ill bump it to a request issue.

To be honest I don't fully understand from your story why you would need a ' prefix'  column (visible? invisible? ) 
Can you give a bit more worked out example? this is not very clear what you want to do, from what I get from your story I don't see any issue, so I must be missing something.

Can you elaborate on:

> But there is a problem about how to deliver the prefix after I type the query string.

If ROFI_DATA provides a solution,  then I see no point of adding something that is already fixed.
